addons/l10n_mx/models/account_account.py

```diff
@@ -17,3 +26,23 @@ def create(self, vals_list):
             tag_id = debit_tag.id if account.code[0] in DEBIT_CODES else credit_tag.id
             account.tag_ids = [Command.link(tag_id)]
         return accounts
+
+    @api.constrains('code', 'group_id')
+    def ensure_code_conforms_coa_sat(self):
+        # filtering accounts used in Mexican reports that do not respect the regulation of the SAT (see documentation for more explanation)
+        # six first characters correspond to the group provided, a list is provided by the SAT
+        # last part is up to the user but should match ACCOUNT_ID_PATTERN which is detailed in the header
+        incorrect_accounts = self.filtered(
+            lambda a: a.group_id and a.account_type != 'equity_unaffected' and a.company_id.country_code == 'MX'
```

Comment:
Account groups are assigned to accounts based on the account's `code` and the group's `code_prefix_start`.
So all accounts that have an account group will already have a code that starts with `code_prefix_start`. 

I don't remember why we filtered on `a.group_id` - can you remind me which accounts we wanted to exclude from the checks? Maybe we need a different way of filtering them out.

Reply:
There are a few accounts that do not have a `group_id` (see account 999999), and not all account groups are correct either (an account in group Activos, with a `code_prefix_start` of 1, would be incorrect but still sent through the COA SAT XML).

Reply:
Thanks so much, I think I was a bit confused, but everything makes sense now. :smile: 
Since we only report accounts that have an account group, it makes sense to only check accounts that have an account group.
https://github.com/odoo/enterprise/blob/42fe8b9c4f103d7b47072ca8c79ddd784dd717f2/l10n_mx_reports/models/trial_balance.py#L131-L134